src/main/java/frc/robot/Alignable.java

```diff
@@ -0,0 +1,38 @@
+package frc.robot;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+
+import java.util.function.BiFunction;
+
+public enum Alignable {
+    AMP(0,0,(x,y)-> new Rotation2d(Math.PI / 2)),
+    SPEAKER(0.46,5.67,(x2,y2)-> new Rotation2d(RobotContainer.shouldFlip() ? 0 : Math.PI).plus(new Rotation2d(Math.atan(y2/x2))));
```

Comment:
I think you should "mirror image" the angle for red, not add 180. I'd like to see why this works.

Reply:
for red the robot faces the "0" degree when looking at speaker headon, for blue the robot faces "180" degrees when looking head on. the math works for red but you need to add the 180 for blue to accommodate for the offset. Furthermore, using Rotation2d.add() keeps the range between -PI/2 and PI/2, necessary for the math to work

Reply:
I got confused myself. I am going to write some unit tests a figure out what is going on